modules/widgets/wordpress-post-widget.php

```diff
@@ -27,138 +73,600 @@ public function __construct() {
+	}
+
+
+	/**
+	 * Parse external API response from the site info call and handle errors if they occur.
+	 *
+	 * @param array|WP_Error $service_response The raw response to be parsed.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function parse_site_info_response( $service_response ) {
+
+		/**
+		 * If the service returned an error, we pass it on.
+		 */
+		if ( is_wp_error( $service_response ) ) {
+			return $service_response;
+		}
+
+		/**
+		 * Check if the service returned proper site information.
+		 */
+		if ( ! isset( $service_response->ID ) ) {
+			return new WP_Error(
+				'no_site_info',
+				__( 'Invalid site information returned from remote.', 'jetpack' ),
+				'No site ID present in the response.'
+			);
+		}
+
+		return $service_response;
+	}
+
+	/**
+	 * Format the posts for better storage. Drop all the data that is not used.
+	 *
+	 * @param object $parsed_data Array of posts returned by the APIs.
+	 *
+	 * @return array Formatted posts or an empty array if no posts were found.
+	 */
+	public function format_posts_for_storage( $parsed_data ) {
+
+		$formatted_posts = array();
+
+		/**
+		 * Only go through the posts list if we have valid posts array.
+		 */
+		if ( isset( $parsed_data->posts ) && is_array( $parsed_data->posts ) ) {
+
+			/**
+			 * Loop through all the posts and format them appropriately.
+			 */
+			foreach ( $parsed_data->posts as $single_post ) {
+
+				$prepared_post = array(
+					'title'          => $single_post->title ? $single_post->title : '',
+					'excerpt'        => $single_post->excerpt ? $single_post->excerpt : '',
+					'featured_image' => $single_post->featured_image ? $single_post->featured_image : '',
+					'url'            => $single_post->URL,
+				);
+
+				/**
+				 * Append the formatted post to the results.
+				 */
+				$formatted_posts[] = $prepared_post;
+			}
 		}
 
-		$site_info = json_decode( $response ['body'] );
-		if ( ! isset( $site_info->ID ) ) {
+		return $formatted_posts;
+	}
+
+
+	/**
+	 * Checks if the cron task is enabled or not. If it is not - enable it.
+	 */
+	public static function check_for_cron() {
+		if ( ! wp_next_scheduled( 'display_posts_widget_cron_update' ) ) {
+			wp_schedule_event( time(), 'minutes_10', 'display_posts_widget_cron_update' );
+		}
+	}
+
+	/**
+	 * Main cron code. Updates all instances of the widget.
+	 *
+	 * @return bool
+	 */
+	public function cron_task() {
+
+		$instances_to_update = $this->get_instances_sites();
+
+		/**
+		 * If no instances are found to be updated - stop.
+		 */
+		if ( empty( $instances_to_update ) || ! is_array( $instances_to_update ) ) {
+			return true;
+		}
+
+		foreach ( $instances_to_update as $site_url ) {
+			$this->update_instance( $site_url );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get a list of unique sites from all instances of the widget.
+	 *
+	 * @return array|bool
+	 */
+	public function get_instances_sites() {
+
+		$widget_settings = $this->wp_get_option( 'widget_jetpack_display_posts_widget' );
```

Comment:
What do you think about "hardcoding" this like `$this->wp_get_option( 'widget_' . $this->widget_id )`?